crates/polars-time/src/round.rs

```diff
@@ -38,3 +38,22 @@ impl PolarsRound for DateChunked {
             .into_date())
     }
 }
+
+#[cfg(feature = "dtype-duration")]
+impl PolarsRound for DurationChunked {
+    fn round(&self, every: Duration, offset: Duration, _tz: Option<&Tz>) -> PolarsResult<Self> {
+        let (every, offset) = match self.time_unit() {
+            TimeUnit::Nanoseconds => (every.duration_ns(), offset.duration_ns()),
+            TimeUnit::Microseconds => (every.duration_us(), offset.duration_us()),
+            TimeUnit::Milliseconds => (every.duration_ms(), offset.duration_ms()),
```

Comment:
`duration_*s` are just approximations, and shouldn't be used when rounding by non-fixed durations, such as `'1mo'`

Can we allow truncating/rounding durations, but only by constant-duration offsets?

You can use `Duration.is_constant_duration` to check this, and raise if it's `false`

Reply:
When raising errors from Rust, is there a preference between `panic!`ing or returning a `PolarsError`?

Reply:
in general it would be better to raise an error